### DISCUSSION TITLE
Build and Test / Build and Test / Test using xcodebuild or run fastlane (pull_request)

### DISCUSSION THREAD

**Discussion by zoyagarg** [CS342/.github#49] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

My pull request failed on the test above
<img width="816" alt="Screenshot 2024-01-15 at 10 42 23 PM" src="https://github.com/CS342/.github/assets/125168583/794dd3f8-f1c9-48af-b5cb-4a874746f555">
<img width="1064" alt="Screenshot 2024-01-15 at 10 43 12 PM" src="https://github.com/CS342/.github/assets/125168583/fa6a012a-35b1-4686-9df7-f1c8292c28d8">
<img width="1076" alt="Screenshot 2024-01-15 at 10 44 45 PM" src="https://github.com/CS342/.github/assets/125168583/d68e38d8-4129-466c-be0a-0fe56a04ff20">
<img width="1073" alt="Screenshot 2024-01-15 at 10 45 59 PM" src="https://github.com/CS342/.github/assets/125168583/35924537-100d-4875-b321-76fb89756d21">
<img width="554" alt="Screenshot 2024-01-15 at 10 47 27 PM" src="https://github.com/CS342/.github/assets/125168583/522d4b5a-2f67-4801-86d3-334ccf42c6a0">

### Reproduction

I included screenshots above, but it looks like my XCode did not build correctly?

### Expected behavior

I'm expecting the build to run XCode quick enough. 

### Additional context

The pull request just seemed like it timed out.

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by bauer-andreas** [comment]
Could you attach a link to your pull request or this action run so we can investigate this in more detail? Would help a lot to have more context on this 🚀

Do the Tests run successfully if you execute them locally?

**Comment by zoyagarg** [comment]
Hi! This is the link to my pull request: https://github.com/CS342/2024-Intake/pull/4#issue-2083102050

What does it mean to execute locally vs in another way?

**Reply by vishnuravi** [reply]
Hi Zoya! What Andreas is referring to is running your app's unit and UI tests locally on your computer vs. on the GitHub runner (server). You can do this by opening up your project in Xcode, then clicking Product > Test. It's good practice to do this locally before pushing your code to GitHub to ensure that you catch any issues beforehand.

In your case, it seems like the problem is a file that you have on your computer that's missing on GitHub, so your tests will likely pass locally but fail on GitHub because the file is missing.

**Comment by vishnuravi** [comment]
Hi Zoya,

I reviewed the logs for the action that failed and it looks like the issue is your new `Questionnaire.json`file is missing from your branch on the repository, although it's referenced in the project. You might have accidentally forgotten to add new files when you committed your changes. If you add the missing file from your git client, this error should resolve.

Best,
Vishnu

**Reply by zoyagarg** [reply]
Okay I think I figured out the earlier issue but I'm still failing one test :(

I'm failing the Reuse Compliance Check and can't figure out why? Is it something wrong with my JSON file again?

**Reply by PSchmiedmayer** [reply]
You can search GitHub Discussions about similar issues that other people might have. This one could be helpful: https://github.com/orgs/CS342/discussions/48

**Reply by zoyagarg** [reply]
Thank you so much for the quick response! So I accidentally deleted all my files, so I just started all over again -- how do you create a file and where can I find a similar license file listed in the discussion you linked? For some reason when I click the "link" it gives me a 404 error, so I am having a little trouble. Thank you so much again.

**Reply by PSchmiedmayer** [reply]
You should be able to re-checkout your branch using GitHub desktop to get back to your state of your PR.

Regarding the files: Your repo contains a lot of `*.license` files. E.g. this one: https://github.com/CS342/2024-Intake/blob/main/.firebaserc.license. All of them contain the same content.

**Comment by zoyagarg** [comment]
Hi Vishnu! How do I add the questionnaire to the branch? All I did was that I made sure I was in the task4 branch in my terminal and dragged the file into my Xcode -- am I missing something? Thank you so much for your help!

Best,
Zoya

**Reply by PSchmiedmayer** [reply, marked as answer]
@zoyagarg The file seems to be added by reference and not copied in the main folder or your project, you can see this here: https://github.com/CS342/2024-Intake/pull/4/files#diff-a080340fd1e2aedeacaf616ac4e2fe6cc47c969833346ec7a761713f1e44e23bR146
```
 path = ../../../../Questionnaire.json
```

If you look at your git diff, you can see that there is no `Questionnaire.json` added to the repo, only some changes to other files: https://github.com/CS342/2024-Intake/pull/4/files

When you drag a file to the Xcode project, be sure that you drag it in there and enable the "Copy Items if Needed" checkbox to ensure that it is copied to the folder of your project and can be tracked in git. Once you did so, it will show up in GitHub desktop and you should be able to commit it.

<img width="842" alt="Screenshot 2024-01-16 at 8 59 45 AM" src="https://github.com/CS342/.github/assets/28656495/5e220255-6abd-4d6e-a4c6-2775c99c7c8d">

You can remove the current reference to the `Questionnaire.json` in your project and then re-add it by dragging it in the Xcode project.

In addition: Please respond to post like this using the "answer" function her ein GitHub Discussions (like this message does). It makes the threads shorter and provides the context right under the main post.
